onesignal.php

```diff
@@ -40,6 +40,13 @@
     // Ensure migration is marked as complete after loading V3
     if (!$is_migrated) {
         update_option('onesignal_plugin_migrated', true);
+    } else {
+        if(!$is_new_install && !isset($settings['notification_on_page'])) {
+            // Upgrade within v3 - enable notification_on_page for existing users
+            // This prevents breaking changes for users who were used to v2 behavior
+            $settings['notification_on_page'] = 1;
+            update_option('OneSignalWPSetting', $settings, 'no');
+        }
```

Comment:
I don't think this code as any bearing on `is_migrated`. No matter what version they are upgrading from (v2->v3.1.3 or v3.0->v3.1.3) we want to do this same migration logic for `notification_on_page`.

Reply:
Thanks, I got kind of confused by `is_migrated` since it gets marked as true even if the user starts on v3.

Reply:
Thanks, I got kind of confused by `is_migrated` since it gets marked as `true` even if the user starts on v3.